src/PoCs/KafkaStatusPopover.tsx

```diff
@@ -0,0 +1,98 @@
+import { Popover, ProgressStep, ProgressStepper, Button } from "@patternfly/react-core";
+import React, {
+  FunctionComponent,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+ 
+type KafkaStatusPopoverProps = {
+  shouldStartOpen?: boolean;
+} & KafkaStatusPopoverBodyProps;
+
+export const KafkaStatusPopover: FunctionComponent<KafkaStatusPopoverProps> = ({
+  shouldStartOpen = false,
+  currentState,
+  children,
+}) => {
+  const [isVisible, setIsVisible] = useState(shouldStartOpen);
+
+  return (
+    <Popover
+      headerContent="Creating instance"
+      bodyContent={<KafkaStatusPopoverBody currentState={currentState} />}
+      isVisible={isVisible}
+      shouldOpen={() => setIsVisible(true)}
+      shouldClose={() => setIsVisible(false)}
+    >
+      {children}
+    </Popover>
+  );
+};
+
+type States = "pending" | "provisioning" | "preparing" | "ready" ;
+
+type KafkaStatusPopoverBodyProps = {
+  currentState: States;
+};
+export const KafkaStatusPopoverBody: VoidFunctionComponent<
+  KafkaStatusPopoverBodyProps
+> = ({ currentState }) => {
+  const order: States[] = ["pending", "provisioning", "preparing", "ready"];
+  const currentStep = order.findIndex((v) => v === currentState);
+  console.log(currentStep);
+  return (
+    <div>
+      <p>This will be ready shortly. In the meantime you can create a service account under the <a href="http://www.google.com/">Connections tab.</a> </p>
+      <br></br>
+      <p> {currentStep} of 4 steps completed</p>
+      <ProgressStepper isVertical>
+        <ProgressStep
+          variant={currentStep === 0 ? "info" : "success"}
+          isCurrent={currentState === "pending"}
+          description="Passed quota check and awaiting creation."
```

Comment:
We don't need periods after these statuses because they're not sentences. The only one that is a sentence is the last one "Kafka instance is ready for use." We could shorten that so that it reads similarly to confirmations (like in a toast, even though this one isn't a toast). Example: "Kafka instance ready for use" or just "Ready for use"

Reply:
@abigaeljamie @lboehling This is what the sample that I shared above would look like with two inline alerts (when the instance takes longer than 15 minutes to create).

![Screenshot 2022-01-17 at 12 46 30](https://user-images.githubusercontent.com/89915947/149772211-b5e8877a-88b5-4654-9e73-9b560f0ab50f.png)